Add `/regex-toggle` slash command

Wolfsblvt opened the pull request:
- Add /regex-toggle command, similarly to /extension-toggle
- toggles the state of both global and character-bound scripts
- Update jsdoc being inconsistent

Closes #3613

<!-- Put X in the box below to confirm -->

## Checklist:

- [x] I have read the [Contributing guidelines](https://www.youtube.com/watch?v=e-GwojpaoQI).

Wolfsblvt committed:
Add /regex-toggle slash command

- Add /regex-toggle command, similarly to /extension-toggle
- toggles the state of both global and character-bound scripts
- Update jsdoc being inconsistent

Closes #3613

Wolfsblvt committed:
Merge branch 'staging' into feat-add-regex-toggle-command

Cohee1207 commented on public/scripts/extensions/regex/index.js:
```diff
@@ -639,6 +669,43 @@ jQuery(async () => {
+                name: 'state',
+                description: 'Explicitly set the state of the script (true to enable, false to disable). If not provided, the state will be toggled to the opposite of the current state.',
+                typeList: [ARGUMENT_TYPE.BOOLEAN],
+                enumList: commonEnumProviders.boolean('trueFalse')(),
```
Doc string suggests `on/off` but the enum is `true/false`

Cohee1207 commented on public/scripts/extensions/regex/index.js:
```diff
@@ -417,6 +411,42 @@ function runRegexCallback(args, value) {
+        return;
+    }
+
+    script.disabled = action === 'enable' ? false : action === 'disable' ? true : !script.disabled;
```
Minor: nested ternaries are barely readable. I'd prefer switch even if more verbose.

Cohee1207 commented on public/scripts/extensions/regex/index.js:
```diff
@@ -417,6 +411,42 @@ function runRegexCallback(args, value) {
+    const index = isScoped ? characters[this_chid]?.data?.extensions?.regex_scripts?.indexOf(script) : scripts.indexOf(script);
+
+    await saveRegexScript(script, index, isScoped);
+    if (script.disabled) {
```
Please add a 'quiet' arg

Wolfsblvt replied:
Which part should be quiet? Just the success message, I figure? Not the "no script found" warning, or the warn/errors inside saving?

Cohee1207 replied:
Yes. The success message.

Cohee1207 commented on public/scripts/extensions/regex/index.js:
```diff
@@ -417,6 +411,42 @@ function runRegexCallback(args, value) {
+
+    if (!script) {
+        toastr.warning(t`Regex script '${scriptName}' not found.`);
+        return;
```
Forgot a return value.

Cohee1207 commented on public/scripts/extensions/regex/index.js:
```diff
@@ -417,6 +411,42 @@ function runRegexCallback(args, value) {
+ * /regex-toggle slash command callback
+ * @param {{state: string}} args Named arguments
+ * @param {string} scriptName The name of the script to toggle
+ * @returns {Promise<string>} The regexed string
```
It doesn't return a regex string.

Cohee1207 reviewed (changes requested):
See above.

Wolfsblvt committed:
On review feedback of /regex-toggle

- Add quiet arg to suppress success toast
- Fix return values
- Switch-case instead of nested ternaries
- state uses onOfToggle
